metabase/metabase
# Revamped row chart

Epic: https://github.com/metabase/metabase/issues/24877

#### Feature requests
Closes https://github.com/metabase/metabase/issues/10967
Closes https://github.com/metabase/metabase/issues/4588
Closes https://github.com/metabase/metabase/issues/5855
Closes https://github.com/metabase/metabase/issues/12977
Closes https://github.com/metabase/metabase/issues/20791

#### Bugs
Closes https://github.com/metabase/metabase/issues/20872
Closes https://github.com/metabase/metabase/issues/15535
Closes https://github.com/metabase/metabase/issues/7310
Closes https://github.com/metabase/metabase/issues/12655

## Changes

Replaces the existing Row chart with the newly rebuilt one with more functionality. Compared to the previous implementation it supports:
1) Multi-series charts: multiple metrics, breakouts (https://github.com/metabase/metabase/issues/5855)
2) Stacked charts (https://github.com/metabase/metabase/issues/10967)
3) Custom axes labels (https://github.com/metabase/metabase/issues/20791)
4) Pow/log scales (https://github.com/metabase/metabase/issues/20791)
5) Goal line (https://github.com/metabase/metabase/issues/4588)
6) Data labels (https://github.com/metabase/metabase/issues/12977)

## Screenshot

<img width="1361" alt="Screen Shot 2022-10-21 at 8 06 43 PM" src="https://user-images.githubusercontent.com/14301985/197240380-15ac7b2b-b71a-4735-b4a3-cc907bedce1d.png">

## How to verify

Well, the goal is to verify that all possible cases in all possible contexts work as expected for a visualization similarly to the existing Bar chart. So ideally, we need to ensure that all combinations of below

#### Try different datasets

Positive values only, negative values only, a mix of values. Small and large datasets.

#### Try various visual settings

- Single series
- Multiple series multiple metrics
- Multiple series breakouts
- Column formatting settings
- Change Series order and visibility
- Change Series settings 
- Axes: different scales, labels
- Goal line and all edge cases with it
- Data labels and their formatting

#### Try different contexts

- Query builder
- Collection view
- Dashboard view (+ fullscreen, + dark mode)
- Embedding

#### Interactivity

- Actions in QB: "view these ...", zoom in, breakout by, filter by this value, automatic explorations
- All types of dashboard click behavior
    - Custom url
    - Update filter
    - Link to another question or dashboard with a parameter value propagation
    - Metabase actions

## TODO
